bot/core/paginator.py

```diff
@@ -0,0 +1,421 @@
+import asyncio
+import logging
+import typing as t
+from textwrap import dedent
+
+import discord
+from discord import Embed
+from discord.abc import Messageable
+from discord.ext import commands
+from discord.ext.commands import Context, Paginator
+
+logger = logging.getLogger(__name__)
+
+
+class PaginatorError(Exception):
+    """
+    Exception Class for Custom error PaginatorError.
+    Raised if the bot is unable to create/process a paginator.
+    """
+    pass
+
+
+def check_permissions(channel: Messageable, bot: commands.Bot) -> bool:
+    """
+    Checks if the bot has required permissions to send or process
+    the paginator
+
+    Args:
+        channel (Messageable): The Context
+        bot (commands.Bot): The bot object, used for using bot.user if there is no guild
+
+    Returns:
+        bool: Whether the bot has all of the required permissions
```

Comment:
We don't use this doctring style, more in [`CONTRIBUTING.md/AutoDoc Formatting Directive`](https://github.com/therealdaneel/HotWired-Bot/blob/master/CONTRIBUTING.md#autodoc-formatting-directive)